code/generator/tests_shared.c

```diff
@@ -1618,25 +1618,65 @@ static void GenerateComponentWiseTests( allocatorLinear_t* tempStorage, stringBu
 			}
 		} );
 
-		Gen_GenerateParametricTestsCode_ComponentWise( tempStorage, code, typeInfo, GEN_FUNCTION_NAME_QUAT_SLERP, strings, flags, &(componentWiseTestsData_t) {
-			.parmDefsCount = 3,
-			.parmDefs = (parametricTestDefinitionParm_t[]) {
-				{ typeInfo,   "lhs" },
-				{ typeInfo,   "rhs" },
-				{ scalarType, "t"   }
-			},
+		{
+			Gen_GenerateParametricTestDefinition_Generic( tempStorage, code, typeInfo, strings, flags, &(parametricTestDefinition_t) {
+				.returnType = typeInfo,
+				.funcName = GEN_FUNCTION_NAME_QUAT_SLERP,
+				.parmsCount = 3,
+				.parms = (parametricTestDefinitionParm_t[]) {
+					{ typeInfo, "lhs" },
+					{ typeInfo, "rhs" },
+					{ scalarType, "percent" }
+				},
+			} );
+			
+			typedef struct testFixture_QuatSlerp_t {
+				float32	lhs[4];
+				float32	rhs[4];
+				float32	percent;
+				float32	expectedAnswer[4];
+			} testFixture_QuatSlerp_t;
```

Comment:
@dangmoody - Right, so I kinda just threw this struct in the middle of the method because I had no idea where you'd want me to put it. Please advise :)

Additionally, I swapped "Gen_GenerateParametricTestsCode_ComponentWise" to instead use "Gen_GenerateParametricTestDefinition_Generic" and "Gen_GenerateParametricTestInvokation_Generic". No idea if that was desired, but it certainly made sense while I was making it.

Completely unrelated, why have you spelt Invocation with a "k"? (Invokation)

Reply:
> Right, so I kinda just threw this struct in the middle of the method because I had no idea where you'd want me to put it. Please advise :)
>
> Additionally, I swapped "Gen_GenerateParametricTestsCode_ComponentWise" to instead use "Gen_GenerateParametricTestDefinition_Generic" and "Gen_GenerateParametricTestInvokation_Generic". No idea if that was desired, but it certainly made sense while I was making it.

I don't see the need to make this a generic test.  I would prefer making as many of the tests component-wise as we possibly can.

I'm a fan of using online calculators as references for what the outputs should be.  If you put in quaternions of `(0, 0, 0, 0)` and `(1, 1, 1, 1)` and a `t` value of `0.25` (for example) into an online quaternion slerp calculator you'll see that the components for the resultant quaternion are all the same value, therefore `slerp` can qualify as a component-wise test.

https://www.euclideanspace.com/maths/algebra/realNormedAlgebra/quaternions/slerp/index.htm

Also this means we can use slightly nicer looking values in tests than some arbitrary quaternions that you grabbed from an animation system or something which won't mean anything to anyone.

---

> Completely unrelated, why have you spelt Invocation with a "k"? (Invokation)

I wasn't aware there was an alternate spelling.

Reply:
> I don't see the need to make this a generic test. I would prefer making as many of the tests component-wise as we possibly can.
> 
> I'm a fan of using online calculators as references for what the outputs should be. If you put in quaternions of `(0, 0, 0, 0)` and `(1, 1, 1, 1)` and a `t` value of `0.25` (for example) into an online quaternion slerp calculator you'll see that the components for the resultant quaternion are all the same value, therefore `slerp` can qualify as a component-wise test.

Okay. The component-wise code isn't as nice to write and I wasn't entirely certain what everything meant. It was also 4am and I wasn't concerned with looking deeply into it. I can change it back to the old version.



> Also this means we can use slightly nicer looking values in tests than some arbitrary quaternions that you grabbed from an animation system or something which won't mean anything to anyone.

Real world examples provide real world results that help us catch bugs. These particular ones helped reproduce the bug and were not chosen arbitrarily. I will adjust these to simplified versions as my previous slerp values were, and capture the 0.25f and 0.75f slerp points for the test.

Reply:
CL 080a13f0447299c91cc21bc5a68a36b52d8f9eb6 addresses these points

Reply:
> > Your changes to the build scripts look unnecessary. It just looks make you've moved the file extension handling to some other part of the script? I don't see the point of this.
> 
> Regarding the .bat scripts, it was passing the .exe in as the output_filename, which results in doubling up on the extensions. For example, the pdb files would appear as _name_.exe.pdb, which seems non-standard to me. These changes fix that.
> 
> The changes to premake5.lua are necessary as the old values are just incorrect and VS won't compile without them. See the updated "source" arguments being passed to build_msvc.bat, for example. It was passing in files that no longer exist (renamed).

I should've mentioned that I'm not disagreeing with the `premake.lua` changes - thank you for patching that up (I don't really touch VS these days).

Also I've just repro'd the file extension problems you mentioned with the build scripts.  I _never_ noticed that.  Nice catch!